3_visualize/src/graph_functions.R

```diff
@@ -0,0 +1,70 @@
+#' @description generates data visualization from model RMSE data
+#' @param outPathdir - the filepath for the output directory
+#' @param eval_data - model results dataframe
+
+plot_data <- function(outPathdir, eval_data){
```

Comment:
Right now your plot function does two things instead of one: it plots the data _and_ it calculates/writes the summary statistics. These should be broken up into more than one function.

Reply:
Agreed. It would be cleaner. Submitting changes.

Reply:
Y'know what? For you, I also say "to heck!" This is for two reasons: (1) you have much less time available to complete this training that the typical onboarder; (2) my personal thought on what you should be getting out of this training is a hands-on overview of how targets works. I'm not sure what your day-to-day looks like. If you're going to be building pipelines from scratch I would suggest doing it; if not, I think you can skip it (I, for example, do a ton of code review).

If you were a jr data scientist I would insist on this change because this is a big anti-pattern that I see in early career staff _and_ on project scale work when working with researchers.

I have documented the need to review this piece here: https://github.com/padilla410/ds-pipelines-targets-1-course-static/issues/8

Reply:
I totally agree with you on the point with incoming data scientists. I know I would have benefited from a similar process earlier in my career. For now, I don't think I'll be doing much code development and any analytics work I have in the near term will likely be modification at best. The bulk of my work lately has been admin related or strategic planning/dev (like the CoE and the QA project). Thanks so much for the thorough review!